### PR TITLE
feat: Add TS types for rolling window

### PR DESCRIPTION
**Pull request by addityasingh**
feat: Add TS types for rolling window

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+type TimeWindow = number; // e.g. 60 * 1000
+type Snapshot = object;
+
+export interface RollingWindowOptions {
+    numChunks: number;
```
Comment (shuhei) [inline index.d.ts]: Each of the options should be optional. For example `{ numChunks: 10 }` should be allowed as valid `options`.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+type TimeWindow = number; // e.g. 60 * 1000
+type Snapshot = object;
```
Comment (shuhei) [inline index.d.ts]: `Snapshot` type needs to have methods, `reset()` and `add(snapshot: Snapshot)`. Could you put these methods?

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+type TimeWindow = number; // e.g. 60 * 1000
+
+interface Snapshot {
```
Comment (shuhei) [inline index.d.ts]: Sorry, I missed this in the first comments (I forgot the context...). This `Snapshot` is more like `Histogram`. The `getSnapshot()` method aggregates multiple histograms in the rolling window and returns a snapshot histogram.

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Reset the snapshot
+     */
+    reset();
+
+    /**
+     * 
+     * @param snapshot Append chunks of snapshot to the snapshot
+     */
+    add(snapshot: Snapshot);
```
Comment (shuhei) [inline index.d.ts]: This is also `add(histogram: Histogram)`

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Record value for the current window
+     */
+    recordValue(value: number);
+
+    /**
+     * Get the snapshot for the current window
+     * @param snapshot Optional snapshot if provided
+     */
+    getSnapshot(snapshot: Snapshot): Snapshot;
```
Comment (shuhei) [inline index.d.ts]: This is something like?

```ts
// @param base Optional histogram to be used to aggregate histograms into if provided
getSnapshot(base?: Histogram): Histogram;
```

Comment (addityasingh) [reply index.d.ts]: I used `AbstractHistogram` from `hdr-histogram-js` instead to better reflect the exact type

Comment (shuhei) [reply index.d.ts]: Ah, that's a good idea!

**Comment by shuhei** [comment]
@addityasingh Thanks a lot! I'll publish this tonight.

**Comment by shuhei** [comment]
@addityasingh Published 0.1.3. Thanks again!

**Pull request merged**